apps/flame-defi/app/features/evm-wallet/services/erc-20-service/index.ts

```diff
@@ -36,60 +36,52 @@ export class Index extends GenericContractService {
   /**
    * Approve a contract to spend tokens on behalf of the user.
    *
-   * @param chainId - The chain ID of the EVM chain
-   * @param contractAddress - The address of the contract that is granted approval to be a recipient of transfers from this ERC20
-   * @param tokenApprovalAmount - The amount of tokens to approve as a string
-   * @param decimals - The number of decimals of the token
-   * @returns Object containing transaction hash if successful
+   * @param chainId - The chain ID of the EVM chain.
+   * @param spender - The address of the contract that is granted approval to be a recipient of transfers from this ERC20.
+   * @param amount - The amount of tokens to approve.
+   * @returns Object containing transaction hash if successful.
    */
-  async approveToken(
+  async approve(
     chainId: number,
-    contractAddress: Address,
-    tokenApprovalAmount: string,
-    decimals: number,
+    spender: Address,
+    amount: bigint,
   ): Promise<Hash> {
-    const amountAsBigInt = parseUnits(tokenApprovalAmount, decimals);
     return await this.writeContractMethod(chainId, "approve", [
-      contractAddress,
-      amountAsBigInt,
+      spender,
+      amount,
     ]);
   }
 
   /**
    * Get the current allowance of a token for a specific contract.
    *
-   * @param chainId - The chain ID of the EVM chain
-   * @param userAddress - The address of the user
-   * @param contractAddress - The address of the contract to check allowance for
-   * @returns The current allowance of the token as a Big.js instance
+   * @param chainId - The chain ID of the EVM chain.
+   * @param owner - The address of the user.
+   * @param spender - The address of the contract to check allowance for.
+   * @returns The current allowance of the token, or null if no allowance.
    */
-  async getTokenAllowance(
+  async allowance(
     chainId: number,
-    userAddress: Address,
-    contractAddress: Address,
-  ): Promise<string | null> {
-    const currentAllowance = await this.readContractMethod(
+    owner: Address,
+    spender: Address,
+  ): Promise<bigint> {
+    const allowance = await this.readContractMethod<bigint>(
       chainId,
       "allowance",
-      [userAddress, contractAddress],
+      [owner, spender],
     );
 
-    if (currentAllowance === null) {
-      return null;
-    }
-
-    const allowance = currentAllowance as bigint;
-    return allowance.toString();
+    return allowance;
   }
 
-  async getBalance(chainId: number, address: string): Promise<bigint> {
+  async balanceOf(chainId: number, address: string): Promise<bigint> {
     return this.readContractMethod(chainId, "balanceOf", [address]);
   }
 }
 
-export function createErc20Service(
-  wagmiConfig: Config,
-  contractAddress: Address,
-): Index {
-  return new Index(wagmiConfig, contractAddress);
+export function createERC20Service(
```

Comment:
The rest of the codebase uses `Erc`

Reply:
There's also `Nft`  and `Evm`. I think we were good about being consistent with naming here previously but now have different patterns